distributed/utils.py

```diff
@@ -477,31 +477,49 @@ def silence_logging(level, root='distributed'):
             logger.setLevel(level)
 
 
-@memoize
-def ensure_ip(hostname):
-    """ Ensure that address is an IP address
 
-    Examples
-    --------
-    >>> ensure_ip('localhost')
-    '127.0.0.1'
-    >>> ensure_ip('123.123.123.123')  # pass through IP addresses
-    '123.123.123.123'
-    """
-    # Prefer IPv4 over IPv6, for compatibility
-    families = [socket.AF_INET, socket.AF_INET6]
-    for fam in families:
+class Ensure_ip(object):
```

Comment:
I recommend camel case for classes

https://www.python.org/dev/peps/pep-0008/#class-names

Reply:
Yes, but in this case Ensure_ip is a callable class masquerading as a function, and thus keeps us from needing to change the calls to ensure_ip throughout the rest of the project, plus changing it back if the upstream issue at Tools/memoize progresses...

Reply:
To be explicit, all I'm asking here is renaming the class to `class EnsureIP(object)` to align with PEP8.  Function name consistency should be maintained by the `ensure_ip` instance that you define just below.

Reply:
Doh. Fixed.